proparse/src/main/java/org/prorefactor/proparse/antlr4/Lexer.java

```diff
@@ -518,13 +537,13 @@ ProToken digitStart() throws IOException {
         case '/':
           append();
           getChar();
-          if (ttype == ProParserTokenTypes.NUMBER)
-            ttype = ProParserTokenTypes.LEXDATE;
+          if (ttype == PreprocessorParser.NUMBER)
+            ttype = PreprocessorParser.LEXDATE;
           break;
         case '\\':
           append();
           getChar();
-          ttype = ProParserTokenTypes.FILENAME;
+          ttype = PreprocessorParser.FILENAME;
           break;
         case '.':
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 6 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1151)